pkgs/development/interpreters/emilua/default.nix

```diff
@@ -90,12 +92,16 @@ stdenv.mkDerivation rec {
     cp "packagefiles/trial.protocol/meson.build" "trial-protocol/"
     popd
 
-    substituteInPlace src/emilua_gperf.awk  --replace '#!/usr/bin/env -S gawk --file' '#!${gawk}/bin/gawk -f'
+    substituteInPlace src/emilua_gperf.awk  --replace '#!/usr/bin/env -S gawk --file' '#!${lib.getExe gawk} -f'
+  '';
```

Comment:
Use `--prefix` flag for each path you add to `EMILUA_PATH`, idem for `NIX_PROFILES`.

I'd recommend you make a temporary wrap flag array then add to the command like `${"$"}{thatFlagArray[@]}`.

Reply:
To be honest I just copied and modified [this line](https://github.com/NixOS/nixpkgs/blob/5d6e3342f2f7c72905b948aca68dda5109260775/pkgs/development/libraries/gstreamer/core/default.nix#L103) in the gstreamer package

Reply:
Expose this postfix in passthru like python does, for example, with `pkgs.python3.sitePackages`

Then reference that variable here.

Simpler and will help a lot when someone else will build a module scope for it.

Reply:
Can you give me a example please?
This line is responsible for the plugins for the interpreter, i will release then soon this package is approved.

Reply:
Something like

```nix
stdenv.mkDerivation (finalAttrs: {

...

    passthru.sitePackages = "/lib/emilua-${(with lib; concatStringsSep "." (take 2 (splitVersion version)))";

...

    --run '${lib.concatStringsSep ":" (map (profile: "${profile}${finalAttrs.passthru.sitePackages))}") $NIX_PROFILES)}'

....
})

```

This approach will not work because NIX_PROFILES is a bash variable, you can't pull that into Nix, but I think you understand the idea.

Reply:
BTW I can follow up this and make tests work

Reply:
Oh, i think i understand, but what the problem with the gstreamer method? other packages use it and works just fine.

Reply:
It works but it's a style thing

Reply:
I will work on this later.